src/libraries/System.Formats.Tar/src/System/Formats/Tar/GnuTarEntry.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Formats.Tar
+{
+    /// <summary>
+    /// Represents a tar entry from an archive of the GNU format.
+    /// </summary>
+    /// <remarks>Even though the <see cref="TarFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in the POSIX IEEE P1003.1 standard from 1988: <c>devmajor</c>, <c>devminor</c>, <c>gname</c> and <c>uname</c>.</remarks>
+    public sealed class GnuTarEntry : PosixTarEntry
+    {
+        // Constructor used when reading an existing archive.
+        internal GnuTarEntry(TarHeader header, TarReader readerOfOrigin)
+            : base(header, readerOfOrigin)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="GnuTarEntry"/> instance with the specified entry type and entry name.
+        /// </summary>
+        /// <param name="entryType">The type of the entry.</param>
+        /// <param name="entryName">A string with the relative path and file name of this entry.</param>
+        /// <exception cref="ArgumentException"><paramref name="entryName"/> is null or empty.</exception>
+        /// <exception cref="InvalidOperationException">The entry type is not supported for creating an entry.</exception>
+        /// <remarks>When creating an instance using the <see cref="GnuTarEntry(TarEntryType, string)"/> constructor, only the following entry types are supported:
+        /// <list type="bullet">
+        /// <item>In all platforms: <see cref="TarEntryType.Directory"/>, <see cref="TarEntryType.HardLink"/>, <see cref="TarEntryType.SymbolicLink"/>, <see cref="TarEntryType.RegularFile"/>.</item>
+        /// <item>In Unix platforms only: <see cref="TarEntryType.BlockDevice"/>, <see cref="TarEntryType.CharacterDevice"/> and <see cref="TarEntryType.Fifo"/>.</item>
+        /// </list>
+        /// </remarks>
+        public GnuTarEntry(TarEntryType entryType, string entryName)
+            : base(entryType, entryName, TarFormat.Gnu)
+        {
+        }
+
+        /// <summary>
+        /// A timestamp that represents the last time the file represented by this entry was accessed.
+        /// </summary>
+        /// <remarks>In Unix platforms, this timestamp is commonly known as <c>atime</c>.</remarks>
+        public DateTimeOffset AccessTime
+        {
+            get => _header._aTime;
+            set
+            {
+                if (value < DateTimeOffset.UnixEpoch)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
```

Comment:
Is there no upper bound on what's supported?  i.e. everything will "just work" if I set this to DateTimeOffset.MaxValue?

Reply:
Would it make sense to set `DateTimeOffset.Now` as the upper bound? Is there precedence of us allowing the user to set a file timestamp to sometime in the future?

Reply:
Another option:

The mtime, atime and ctime fields that are stored directly in the header (not in extended attributes) have a limit of 12 bytes, and are stored as octal integers. The maximum number we can store in such location is `777,777,777,777` in octal, or `68,719,476,735` in decimal.

Reply:
Actually, no: The `mtime` field has to end with a space character, according to the spec. So we only have 11 bytes available for the number. That means the max number in octal is `77,777,777,777` or `8,589,934,591` in decimal. I'll adjust this.

Reply:
I may have to revisit this later. I think the upper bound will cause trouble in V7, Ustar and Gnu, but not in PAX: The advantage of PAX having extended attributes is that we can store values with no length limit. One of the rules is that if we find out that mtime is too large to fit in 12 bytes, we save it as an extended attribute with its full value, and in the header field we only write what fits. Then, when the user needs to consult the `ModificationTime` property, we need to return the correct value we retrieved from the extended attribute, not from the header.

Reply:
Re future timestamps: they are sometimes useful. Eg when we shipped product on CD/DVD we generally would set the file times to a particular future value, so it was clear they all belonged together. Another hypothetical scenario is that you're on a device whose clock is not reliable.

Reply:
Checking current time is expensive. IMO the current implementation (having only the `UnixEpoch` check) is all we need.